include/lsst/afw/image/TransmissionCurve.h

```diff
@@ -0,0 +1,300 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST/AURA.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_IMAGE_TransmissionCurve_h_INCLUDED
+#define LSST_AFW_IMAGE_TransmissionCurve_h_INCLUDED
+
+#include "ndarray_fwd.h"
+
+#include "lsst/afw/geom/Transform.h"
+#include "lsst/afw/geom/Box.h"
+#include "lsst/afw/table/io/Persistable.h"
+
+namespace lsst {
+namespace afw {
+namespace image {
+
+
+/**
+ *  A spatially-varying transmission curve as a function of wavelength.
+ *
+ *  TransmissionCurve can only be evaluated at discrete user-provided positions;
+ *  it does not provide an interface for computing average transmission over
+ *  regions or computing spatially-varying scalars from integrals over the
+ *  wavelength dimension.
+ *
+ *  TransmissionCurves are immutable and are expected to be passed and held
+ *  by shared_ptr.  As such they are neither copyable nor movable.
+ *
+ *  All wavelength values should be in Angstroms.
+ *
+ *  The flux units and overall normalization of TransmissionCurves is
+ *  unspecified by the class, but their normalization and units should always
+ *  be consistent throughout the spatial area over which they are defined
+ *  (an implementation should not e.g. re-normalize to unit bolometric flux
+ *  at each position it is evaluated at).  Other classes and functions using
+ *  TransmissionCurves should of course document the flux units and/or
+ *  normalization expected/provided.
+ */
+class TransmissionCurve : public table::io::PersistableFacade<TransmissionCurve>,
+                          public table::io::Persistable
+{
+public:
+
+    /**
+     *  A simple struct used to specify the wavelength grid for evaluation.
+     */
+    struct SampleDef {
+
+        /// Construct with infinite bounds and zero size.
+        SampleDef();
+
+        /// Construct with the given bounds and size.
+        SampleDef(double min_, double max_, int size_) : min(min_), max(max_), size(size_) {}
```

Comment:
I don't think the style guide specifies the any way to deal with this case (where the data members are public and hence do not have leading underscores).  This seemed as good an approach as any given the lack of guidance.

Reply:
And you don't want to just call them `min`, `max`, and `size`? IIRC the code would work.

Reply:
I strongly agree that just calling them min, max and size is the best solution. It works fine and it's a lot less ugly and surprising than adding underscores or otherwise munging names.

Reply:
The code would indeed work, but I think shadowing variables is sufficiently bad idea in general that it should be avoided even in simple cases.